error/s2n_errno.c

```diff
@@ -14,155 +14,181 @@
+    ERROR_STRING(S2N_ERR_UNRECOGNIZED_EXTENSION, "TLS extension not recognized" )
+    ERROR_STRING(S2N_ERR_INVALID_SCT_LIST, "SCT list is invalid" )
+    ERROR_STRING(S2N_ERR_INVALID_OCSP_RESPONSE, "OCSP response is invalid" )
+    ERROR_STRING(S2N_ERR_UPDATING_EXTENSION, "Updating extension data failed" )
+    ERROR_STRING(S2N_ERR_INVALID_NONCE_TYPE, "Invalid AEAD nonce type")
+    ERROR_STRING(S2N_ERR_UNIMPLEMENTED, "Unimplemented feature")
+    ERROR_STRING(S2N_ERR_READ, "error calling read")
+    ERROR_STRING(S2N_ERR_WRITE, "error calling write")
+    ERROR_STRING(S2N_ERR_CERT_UNTRUSTED, "Certificate is untrusted")
+    ERROR_STRING(S2N_ERR_CERT_TYPE_UNSUPPORTED, "Certificate Type is unsupported")
+    ERROR_STRING(S2N_ERR_CANCELLED, "handshake was cancelled")
+    ERROR_STRING(S2N_ERR_INVALID_MAX_FRAG_LEN, "invalid Maximum Fragmentation Length encountered")
+    ERROR_STRING(S2N_ERR_MAX_FRAG_LEN_MISMATCH, "Negotiated Maximum Fragmentation Length from server does not match the requested length by client")
+    ERROR_STRING(S2N_ERR_INVALID_SERIALIZED_SESSION_STATE, "Serialized session state is not in valid format")
+    ERROR_STRING(S2N_ERR_SERIALIZED_SESSION_STATE_TOO_LONG, "Serialized session state is too long")
+    ERROR_STRING(S2N_ERR_SESSION_ID_TOO_LONG, "Session id is too long")
+    ERROR_STRING(S2N_ERR_CLIENT_AUTH_NOT_SUPPORTED_IN_SESSION_RESUMPTION_MODE, "Client Auth is not supported in session resumption mode")
+    ERROR_STRING(S2N_ERR_INVALID_TICKET_KEY_LENGTH, "Session ticket key length cannot be zero")
+    ERROR_STRING(S2N_ERR_INVALID_TICKET_KEY_NAME_OR_NAME_LENGTH, "Session ticket key name should be unique and the name length cannot be zero")
+    ERROR_STRING(S2N_ERR_TICKET_KEY_NOT_UNIQUE, "Cannot add session ticket key because it was added before")
+    ERROR_STRING(S2N_ERR_TICKET_KEY_LIMIT, "Limit reached for unexpired session ticket keys")
+    ERROR_STRING(S2N_ERR_NO_TICKET_ENCRYPT_DECRYPT_KEY, "No key in encrypt-decrypt state is available to encrypt session ticket")
+    ERROR_STRING(S2N_ERR_ENCRYPT_DECRYPT_KEY_SELECTION_FAILED, "Failed to select a key from keys in encrypt-decrypt state")
+    ERROR_STRING(S2N_ERR_KEY_USED_IN_SESSION_TICKET_NOT_FOUND, "Key used in already assigned session ticket not found for decryption")
+    ERROR_STRING(S2N_ERR_SENDING_NST, "Error in session ticket status encountered before sending NST")
+    ERROR_STRING(S2N_ERR_INVALID_DYNAMIC_THRESHOLD, "invalid dynamic record threshold")
+    ERROR_STRING(S2N_ERR_INVALID_ARGUMENT, "invalid argument provided into a function call")
+    ERROR_STRING(S2N_ERR_NOT_IN_UNIT_TEST, "Illegal configuration, can only be used during unit tests")
 };
+const int num_of_errors = sizeof(S2N_ERROR_EN) / sizeof(S2N_ERROR_EN[0]);
+static struct s2n_map *error_translation_table = NULL;
+
+static s2n_error_translation *s2n_lookup_error_translation(int error)
+{
+    struct s2n_blob k, v;
+    char key[KEY_BUF_LEN];
+    snprintf(key, KEY_BUF_LEN, "%08x", error);
```

Comment:
Just use the int as the key directly :)  Something like ... 

union {
    char bytes[sizeof(int)];
    int raw;
} key; 
key.native = error;

...

k.data = (void *) key.bytes;
k.size = sizeof( key.bytes );

Reply:
> Just use the int as the key directly :) Something like ...
> 
> union {
> char bytes[sizeof(int)];
> int raw;
> } key;
> key.native = error;
> 
> ...
> 
> k.data = (void *) key.bytes;
> k.size = sizeof( key.bytes );

That is brilliant idea. I will do that :)